### PR TITLE
Add validation error messages for conditional questions

### PR DESCRIPTION
**Pull request by quis**
https://www.pivotaltracker.com/story/show/109610014

> Would be nice for the condtionally required fields in the declaration to have
> a custom message e.g. "You must answer this question because you answered Yes
> to..."

### CODE REVIEW

**Comment by superroz** [comment]
Cool but need to change Yes to 'yes' to be consistent with other content in the declaration.

**Comment by TheDoubleK** [comment]
I think there are a couple more too - The ones that are required if you're not UK company (i.e. "No" to `establishedInTheUK`) - `appropriateTradeRegisters` and `licenseOrMemberRequired`

File: frameworks/digital-outcomes-and-specialists/questions/declaration/mitigatingFactors2.yml [hunk 1]
```diff
@@ -1,2 +1,4 @@
 question: If you responded yes to either question [[unspentTaxConvictions]] or question [[GAAR]], please provide details of any mitigating factors that you think should be taken into consideration.
 type: textbox_large
+  - name: answer_required
```
Comment (allait) [inline frameworks/digital-outcomes-and-specialists/questions/declaration/mitigatingFactors2.yml]: This is missing the `validations:` line

File: frameworks/digital-outcomes-and-specialists/questions/declaration/appropriateTradeRegisters.yml [hunk 2]
```diff
@@ -1,3 +1,6 @@
 question: If your business was not established in the UK, is it registered with the appropriate professional or trade register(s) in the EU member state where it’s established?
 type: boolean
 hint: Check that your organisation complies with [Schedule 5, Regulation 58(5)](http://www.legislation.gov.uk/uksi/2015/102/regulation/58/made) of the Public Contracts Regulations 2015.
+validations:
+  - name: answer_required
+    message: You need to answer this question because you anwered 'no' to [[establishedInTheUK]].
```
Comment (TheDoubleK) [inline frameworks/digital-outcomes-and-specialists/questions/declaration/appropriateTradeRegisters.yml]: Fancy quotes?
@ralph-hawkins

File: frameworks/digital-outcomes-and-specialists/questions/declaration/licenceOrMemberRequired.yml [hunk 3]
```diff
@@ -5,3 +5,6 @@ options:
   - label: licensed
   - label: a member of a relevant organisation
   - label: none of the above
+validations:
+  - name: answer_required
+    message: You need to answer this question because you anwered 'no' to [[establishedInTheUK]].
```
Comment (TheDoubleK) [inline frameworks/digital-outcomes-and-specialists/questions/declaration/licenceOrMemberRequired.yml]: Fancy quotes?
I have internalised @ralph-hawkins

Comment (quis) [inline frameworks/digital-outcomes-and-specialists/questions/declaration/licenceOrMemberRequired.yml]: I’ll fix it.

**Comment by robyoung** [comment]
:+1:

**Pull request merged**